[BUG] Error in indexer when ordering make on ibetExchange

## Describe the bug
The following error occurred in [ibet-Wallet-API](https://github.com/BoostryJP/ibet-Wallet-API).
refs BoostryJP/ibet-Wallet-API#1099
```
[2022-01-17 02:47:30 +0000] [INDEXER-POSITION-MEMBERSHIP] [37] [INFO] syncing from=2461470, to=2461479                                                                                                                                   │
│   [2022-01-17 02:47:31 +0000] [INDEXER-POSITION-MEMBERSHIP] [37] [ERROR] Class 'builtins.NoneType' is not mapped                                                                                                                           │
│   Traceback (most recent call last):                                                                                                                                                                                                       │
│     File "batch/indexer_Position_Membership.py", line 331, in __sync_exchange                                                                                                                                                              │
│       self.sink.on_position(                                                                                                                                                                                                               │
│     File "batch/indexer_Position_Membership.py", line 67, in on_position                                                                                                                                                                   │
│       sink.on_position(*args, **kwargs)                                                                                                                                                                                                    │
│     File "batch/indexer_Position_Membership.py", line 115, in on_position                                                                                                                                                                  │
│       self.db.merge(position)                                                                                                                                                                                                              │
│     File "<string>", line 2, in merge                                                                                                                                                                                                      │
│     File "/home/apl/.pyenv/versions/3.8.7/lib/python3.8/site-packages/sqlalchemy/orm/session.py", line 2969, in merge                                                                                                                      │
│       object_mapper(instance)  # verify mapped                                                                                                                                                                                             │
│     File "/home/apl/.pyenv/versions/3.8.7/lib/python3.8/site-packages/sqlalchemy/orm/base.py", line 296, in object_mapper                                                                                                                  │
│       return object_state(instance).mapper                                                                                                                                                                                                 │
│     File "/home/apl/.pyenv/versions/3.8.7/lib/python3.8/site-packages/sqlalchemy/orm/base.py", line 318, in object_state                                                                                                                   │
│       raise exc.UnmappedInstanceError(instance)                                                                                                                                                                                            │
│   sqlalchemy.orm.exc.UnmappedInstanceError: Class 'builtins.NoneType' is not mapped
```

## Expected behavior
Being skiped `merge` when registering with 0 holdings.

--- a/batch/indexer_position_bond.py
+++ b/batch/indexer_position_bond.py
@@ -107,6 +107,7 @@ class DBSink:
                 position.exchange_balance = exchange_balance
             if exchange_commitment is not None:
                 position.exchange_commitment = exchange_commitment
+            self.db.merge(position)
         elif any(value is not None and value > 0
                  for value in [balance, pending_transfer, exchange_balance, exchange_commitment]):
             LOG.debug(f"Position created (Bond): token_address={token_address}, account_address={account_address}")
@@ -117,8 +118,7 @@ class DBSink:
             position.pending_transfer = pending_transfer or 0
             position.exchange_balance = exchange_balance or 0
             position.exchange_commitment = exchange_commitment or 0
-
-        self.db.merge(position)
+            self.db.add(position)
 
     def flush(self):
         self.db.commit()
@@ -266,18 +266,10 @@ class Processor:
                     fromBlock=block_from,
                     toBlock=block_to
                 )
-
-                # Get exchange contract address
-                exchange_contract_address = ContractUtils.call_function(
-                    contract=token,
-                    function_name="tradableExchange",
-                    args=(),
-                    default_returns=ZERO_ADDRESS
-                )
                 for event in events:
                     args = event["args"]
                     for account in [args.get("from", ZERO_ADDRESS), args.get("to", ZERO_ADDRESS)]:
-                        if account != exchange_contract_address:
+                        if web3.eth.getCode(account).hex() == "0x":
                             balance, pending_transfer, exchange_balance, exchange_commitment = \
                                 self.__get_account_balance_all(token, account)
                             self.sink.on_position(

--- a/batch/indexer_position_share.py
+++ b/batch/indexer_position_share.py
@@ -107,6 +107,7 @@ class DBSink:
                 position.exchange_balance = exchange_balance
             if exchange_commitment is not None:
                 position.exchange_commitment = exchange_commitment
+            self.db.merge(position)
         elif any(value is not None and value > 0
                  for value in [balance, pending_transfer, exchange_balance, exchange_commitment]):
             LOG.debug(f"Position created (Share): token_address={token_address}, account_address={account_address}")
@@ -117,8 +118,7 @@ class DBSink:
             position.pending_transfer = pending_transfer or 0
             position.exchange_balance = exchange_balance or 0
             position.exchange_commitment = exchange_commitment or 0
-
-        self.db.merge(position)
+            self.db.add(position)
 
     def flush(self):
         self.db.commit()
@@ -266,18 +266,10 @@ class Processor:
                     fromBlock=block_from,
                     toBlock=block_to
                 )
-
-                # Get exchange contract address
-                exchange_contract_address = ContractUtils.call_function(
-                    contract=token,
-                    function_name="tradableExchange",
-                    args=(),
-                    default_returns=ZERO_ADDRESS
-                )
                 for event in events:
                     args = event["args"]
                     for account in [args.get("from", ZERO_ADDRESS), args.get("to", ZERO_ADDRESS)]:
-                        if account != exchange_contract_address:
+                        if web3.eth.getCode(account).hex() == "0x":
                             balance, pending_transfer, exchange_balance, exchange_commitment = \
                                 self.__get_account_balance_all(token, account)
                             self.sink.on_position(

--- a/batch/processor_create_utxo.py
+++ b/batch/processor_create_utxo.py
@@ -238,7 +238,7 @@ class Processor:
                 amount = args.get("value")
 
                 # Skip sinking in case of deposit to exchange or withdrawal from exchange
-                if from_account == exchange_contract_address or to_account == exchange_contract_address:
+                if web3.eth.getCode(from_account).hex() != "0x" or web3.eth.getCode(to_account).hex() != "0x":
                     continue
 
                 transaction_hash = event["transaction_hash"]

